File: Strings/LowerUpper.java
// convert a string to upper or lower case

public static String toLower(String str) {
    String result = "";
    for (int i = 0; i < str.length(); i++) {
        if (str.charAt(i) >= 'A' && str.charAt(i) <= 'Z')
            result += (char) (str.charAt(i) + ('a' - 'A'));
        else
            result += str.charAt(i);
    }
    return result;
}

public static String toUpper(String str) {
    String result = "";
    for (int i = 0; i < str.length(); i++) {
        if (str.charAt(i) >= 'a' && str.charAt(i) <= 'z')
            result += (char) (str.charAt(i) - ('a' - 'A'));
        else
            result += str.charAt(i);
    }
    return result;
}

File: Strings/LowerUpper.js
// convert a string to upper or lower case

// Time Complexity  O(n)
// Space Complexity O(1)

function toLower(str) {
    let result = "";
    for (let i = 0; i < str.length; i++) {
        if (str.charAt(i) >= "A" && str.charAt(i) <= "Z")
            result += String.fromCharCode(str.charCodeAt(i) + 32);
        else result += str.charAt(i);
    }
    return result;
}

function toUpper(str) {
    let result = "";
    for (let i = 0; i < str.length; i++) {
        if (str.charAt(i) >= "a" && str.charAt(i) <= "z")
            result += String.fromCharCode(str.charCodeAt(i) - 32);
        else result += str.charAt(i);
    }
    return result;
}
